src/smclient/smclient.hpp

```diff
@@ -0,0 +1,193 @@
+/*
+ * Copyright (C) 2024 EPAM Systems, Inc.
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#ifndef SMCLIENT_HPP_
+#define SMCLIENT_HPP_
+
+#include <condition_variable>
+#include <thread>
+
+#include <grpcpp/channel.h>
+#include <grpcpp/security/credentials.h>
+
+#include <aos/common/alerts/alerts.hpp>
+#include <aos/common/cloudprotocol/alerts.hpp>
+#include <aos/common/crypto/crypto.hpp>
+#include <aos/common/crypto/utils.hpp>
+#include <aos/common/monitoring/monitoring.hpp>
+#include <aos/common/tools/error.hpp>
+#include <aos/iam/certhandler.hpp>
+#include <aos/iam/nodeinfoprovider.hpp>
+#include <aos/iam/provisionmanager.hpp>
+#include <aos/sm/launcher.hpp>
+#include <aos/sm/logprovider.hpp>
+#include <aos/sm/networkmanager.hpp>
+#include <aos/sm/resourcemanager.hpp>
+
+#include <servicemanager/v4/servicemanager.grpc.pb.h>
+
+#include "config/config.hpp"
+
+namespace SMProto = servicemanager::v4;
+
+namespace aos::sm::client {
+
+using PublicNodeService = SMProto::SMService;
+
+/**
+ * GRPC service manager client.
+ */
+class SMClient : public iam::certhandler::CertReceiverItf,
+                 public monitoring::SenderItf,
+                 public alerts::SenderItf,
+                 public sm::logprovider::LogObserverItf,
+                 public sm::launcher::InstanceStatusReceiverItf,
+                 private NonCopyable {
+public:
+    /**
+     * Initializes SM client instance.
+     *
+     * @param config client configuration.
+     * @param provisionManager provision manager.
+     * @param certLoader certificate loader.
+     * @param cryptoProvider crypto provider.
+     * @param nodeInfoProvider node info provider.
+     * @param resourceManager resource manager.
+     * @param networkManager network manager.
+     * @param logProvider log provider.
+     * @param resourceMonitor resource monitor.
+     * @param launcher launcher.
+     * @param provisioningMode flag indicating whether provisioning mode is active.
+     * @returns Error.
+     */
+    Error Init(const config::Config& config, iam::provisionmanager::ProvisionManagerItf& provisionManager,
+        crypto::CertLoaderItf& certLoader, crypto::x509::ProviderItf& cryptoProvider,
+        iam::nodeinfoprovider::NodeInfoProviderItf& nodeInfoProvider,
+        sm::resourcemanager::ResourceManagerItf& resourceManager, sm::networkmanager::NetworkManagerItf& networkManager,
+        sm::logprovider::LogProviderItf& logProvider, monitoring::ResourceMonitorItf& resourceMonitor,
+        sm::launcher::LauncherItf& launcher, bool provisioningMode);
```

Comment:
remove `provisionedMode` flag. SM always works in normal mode. Probably, you need open connection for tests then it should be renamed and default value should be secure connection.